core/execution_state.cpp

```diff
@@ -132,11 +132,10 @@ void ExecutionState::addReachabilityMapping(const IR::Node *node, const IR::Expr
 
     if (!reachabilityMap.initializeReachabilityMapping(
             node, new IR::LAnd(getExecutionCondition(), cond))) {
-        // Throw a fatal error if we try to add a duplicate mapping.
-        // This can affect the correctness of the entire mapping.
-        BUG("Reachability mapping for node %1% already exists. Every mapping must be uniquely "
-            "identifiable.",
-            node);
+        // We originally throw a fatal error if we try to add a duplicate mapping.
```

Comment:
I am feeling we should add a `--strict`` mode here. This way we also do not have the problem with the warnings.

Reply:
I agree. Will add this